source/val/validation_state.cpp

```diff
@@ -425,6 +426,22 @@ bool ValidationState_t::RegisterUniqueTypeDeclaration(
     key.insert(key.end(), inst.words + words_begin, inst.words + words_end);
   }
 
+  if (inst.opcode == SpvOpTypePointer &&
```

Comment:
Better to have a comment describing the rule you're enforcing.

Reply:
Done.